packages/@sanity/language-filter/src/SelectLanguageProvider.js

```diff
@@ -4,23 +4,31 @@ import {selectedLanguages$, setLangs} from './datastore'
 import SelectLanguage from './SelectLanguage'
 
 export default class SelectLanguageProvider extends React.Component {
-  state = {selected: []}
+  state = {
+    selected: [],
+    currentDocumentType: null,
+  }
 
   componentDidMount(props) {
     this.subscription = selectedLanguages$.subscribe((selected) => {
-      this.setState({selected: selected})
+      this.setState({
+        selected: selected,
+        currentDocumentType: this.props?.schemaType?.name,
```

Comment:
Any reason to track this as component state instead of reading from props at render time?

Reply:
I don't know what you mean, I'm sorry? 

If there is a different way to do this it was wasn't obvious, sorry.

Reply:
Sorry for being unclear! What I'm suggesting is to read `currentDocumentType` from the props in the render function instead of storing it in the component's state, unless there are good reasons to do it this way.

So instead of copying from initial props to state here you can read it from props at render time:
```js
render() {
  const {selected} = this.state
  const currentDocumentType = this.props.schemaType?.name
  //...
}
```
This way it will also respond to changes in the given `type` prop.

Hope that was clearer :)

Reply:
Ah, cool, I see what you mean now - I'll make that change